### PR TITLE
flannel: use rkt to launch flannel on amd64

### PR DESCRIPTION
**Pull request by ajeddeloh**
Launch flannel with rkt instead of an early instance of docker and remove early-docker

### CODE REVIEW

**Comment by marineam** [comment]
Both ebuild revisions need bumping

File: app-admin/flannel/files/flanneld.service [hunk 1]
```diff
@@ -15,28 +13,29 @@ Environment="FLANNEL_VER={{flannel_ver}}"
 Environment="FLANNEL_IMG=quay.io/coreos/flannel"
 Environment="ETCD_SSL_DIR=/etc/ssl/etcd"
 Environment="FLANNEL_ENV_FILE=/run/flannel/options.env"
```
Comment (marineam) [inline app-admin/flannel/files/flanneld.service]: Is FLANNEL_ENV_FILE still needed?

File: app-admin/flannel/files/flanneld.service [hunk 2]
```diff
@@ -15,28 +13,29 @@ Environment="FLANNEL_VER={{flannel_ver}}"
...
-  --env-file=${FLANNEL_ENV_FILE} \
-  --volume=/usr/share/ca-certificates:/etc/ssl/certs:ro \
-  --volume=${ETCD_SSL_DIR}:${ETCD_SSL_DIR}:ro \
-  ${FLANNEL_IMG}:${FLANNEL_VER} /opt/bin/flanneld --ip-masq=true
+ExecStart=/usr/bin/rkt run --net=host \
+   --stage1-path=/usr/lib/rkt/stage1-images/stage1-fly.aci \
+   --insecure-options=image \
+   --set-env=NOTIFY_SOCKET=/run/systemd/notify \
+   --set-env=AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} \
+   --set-env=AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} \
```
Comment (marineam) [inline app-admin/flannel/files/flanneld.service]: If inherit-env is true why do these vars need to be duplicated?

Comment (ajeddeloh) [inline app-admin/flannel/files/flanneld.service]: Good point, fixed.

File: app-admin/flannel/files/flanneld.service (hunk 1 above)
Comment (ajeddeloh) [inline app-admin/flannel/files/flanneld.service]: Nope, and neither is DOCKER_HOST. Fixed.

**Comment by glevand** [comment]
We'll need https://github.com/coreos/rkt/pull/2758 for this to work for arm64.

**Comment by marineam** [comment]
Looks ok, though a thought so this can go forward without impacting arm while we wait on rkt there: leave early docker in place and change the flannel commit to create a new ebuild reversion and service file, keeping the old available. The new ebuild should only be keyworded on amd64 and use the new rkt service.

**Comment by ajeddeloh** [comment]
Bump, added new ebuild for amd64 only, left early-docker alone

File: app-admin/flannel/flannel-9999.ebuild [hunk 3]
```diff
@@ -23,7 +23,7 @@ RDEPEND="app-admin/sdnotify-proxy"
 S="$WORKDIR"
```
Comment (marineam) [inline app-admin/flannel/flannel-9999.ebuild]: Remove `RDEPEND="app-admin/sdnotify-proxy"` since it is no longer used. Set to rkt instead.

Comment (ajeddeloh) [inline app-admin/flannel/flannel-9999.ebuild]: Fixed.

Comment (marineam) [inline app-admin/flannel/flannel-9999.ebuild]: I don't see it

Comment (ajeddeloh) [inline app-admin/flannel/flannel-9999.ebuild]: woops, my bad. there now

**Comment by marineam** [comment]
LGTM

**Comment by glevand** [comment]
Seems like that will work.

**Pull request merged**